### PR TITLE
Fixes #18333 katello-backup returns 0 exit code even when failing

### PR DESCRIPTION
**Pull request by jcpunk**
Checking the success of the various backup operations.  Aborts the backup with a hopefully useful error message on failure.

### CODE REVIEW

**Comment by stbenjam** [comment]
I'd extract all the logic to a method that executes a command, and then fails with a message if the command doesn't succeed instead of having the same block of code repeated throughout the program.

**Comment by jcpunk** [comment]
Something like this last commit?

**Comment by ehelms** [comment]
@stbenjam could you take another look at this, thanks!

**Comment by cfouant** [comment]
@stbenjam ping, could you re-review pls? Thanks!

**Comment by cfouant** [comment]
@jcpunk could you rebase (you might need to do it again, depending on which of your PRs merges next). Thanks!

**Comment by jcpunk** [comment]
:) Can do, rebased

**Comment by stbenjam** [comment]
[test]

File: katello/katello-backup [hunk 1]
```diff
@@ -40,14 +40,22 @@ optparse = OptionParser.new do |opts|
   @dir = ARGV[0].dup
 end
 
+def run_cmd(command)
+  `#{command}`
+  if $?.exitstatus != 0
+    puts "Failed '#{command}'"
```
Comment (stbenjam) [inline katello/katello-backup]: Overall the change looks good to me, I'd prefer if we did `STDERR.puts` here instead (and everywhere else we print an error and exit).

Some sysadmins have cron send stdout to /dev/null, but keep stderr so they get a mail from cron when the job fails that has the error message in it.

**Comment by jcpunk** [comment]
Updated to use `STDERR.puts`

**Comment by stbenjam** [comment]
👍 
Now just needs a rebase since https://github.com/Katello/katello-packaging/pull/389 was just merged

**Comment by jcpunk** [comment]
rebased

**Comment by ehelms** [comment]
I am taking @stbenjam thumbs up as approval since this looks good I am going to merge.

**Comment by cfouant** [comment]
Hey guys - this actually breaks katello-backup. Any way we can revert this merge?

File: katello/katello-backup [hunk 2]
```diff
@@ -44,14 +44,22 @@ optparse = OptionParser.new do |opts|
   @dir = ARGV[0].dup
 end
 
+def run_cmd(command)
+  `#{command}`
+  if $?.exitstatus != 0
```
Comment (cfouant) [inline katello/katello-backup:49]: The config file backup will generally return exitstatus 2, as the file systems are different for puppet 3 vs puppet 4, which means some files that are called in the compression don't exist, based on which puppet is being used.

Comment (jcpunk) [reply katello/katello-backup:49]: If you've got error snippets, I can make it smarter.  I'd rather fix the check than defeat it.

Something like -> https://github.com/jcpunk/katello-packaging/blob/puppet_conf_3_vs_4/katello/katello-backup ?

Comment (ehelms) [reply katello/katello-backup:49]: Your branch has what @cfouant and I were discussing as a possible alternative form.

Comment (ehelms) [reply katello/katello-backup:49]: @jcpunk can you open a PR with your proposed update?

Comment (jcpunk) [reply katello/katello-backup:49]: Done -> #405

**Pull request merged**